### PR TITLE
first math task. Mark

### PR DESCRIPTION
**Pull request by markru37**

### CODE REVIEW

File: CourseApp/Program.cs [hunk 1]
```diff
@@ -6,8 +6,61 @@ public class Program
...
+                Console.WriteLine($"x={x}, y={y}");
+            }
+        }
+
+        public static double CalcFunc(double a, double b, double x)
+        {
+            var numerator = Math.Pow(Math.Pow(x - a, 2), 1 / 3.0) + Math.Pow(Math.Abs(x + b), 1 / 5.0);
+            var denominator = Math.Pow(Math.Pow(x, 2) - Math.Pow(a - b, 2), 1 / 9.0);
+
+            var y = numerator / denominator;
```
Comment (alexthvest) [inline CourseApp/Program.cs:34]: Looks weird

File: CourseApp/Program.cs [hunk 2]
```diff
@@ -6,8 +6,61 @@ public class Program
...
         {
-            Console.WriteLine($"Hello world");
-            Console.ReadLine();
+            const double a = 0.8;
+            const double b = 0.4;
+            Console.WriteLine($"--------- TASK A --------------");
+            var taskA = TaskA(a, b, 1.23, 7.23, 1.2);
+            foreach (var i in taskA)
+            {
+                var (x, y) = i;
```
Comment (alexthvest) [inline CourseApp/Program.cs:15]: You can destruct it in foreach loop

File: CourseApp/Program.cs [hunk 3]
```diff
@@ -6,8 +6,56 @@ public class Program
...
+                res[i] = (x, y);
+                i++;
+            }
+
+            return res;
+        }
+
+        public static (double x, double y)[] TaskB(double a, double b, double[] xArray)
+        {
+            var res = new (double, double)[xArray.Length];
```
Comment (alexthvest) [inline CourseApp/Program.cs:48]: Please, use List<T>

File: CourseApp/Program.cs [hunk 4]
```diff
@@ -6,8 +6,56 @@ public class Program
...
+            int i = 0;
```
Comment (alexthvest) [inline CourseApp/Program.cs:49]: Why we have `i` and `i1`?

File: CourseApp/Program.cs (hunk 3 above)
Comment (jskonst) [reply CourseApp/Program.cs:48]: We did not yet managed to work with lists... but I guess you already know how to do it :)

**Comment by jskonst** [comment]
@markru37 Please resolve merge conflict

File: CourseApp/Calc.cs [hunk 5]
```diff
@@ -0,0 +1,81 @@
+namespace CourseApp
+{
+    using System;
+    using System.Collections.Generic;
+    using Kantaiko.ConsoleFormatting;
+
+    public class Calc
```
Comment (alexthvest) [inline CourseApp/Calc.cs:7]: Please, do not shorter names

File: CourseApp/Calc.cs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        {
+            var a = 0.8;
+            var b = 0.4;
+            var xs = 1.23;
+            var xe = 7.23;
+            var dx = 1.2;
+            var listA = new List<double>();
+            var listB = new List<double>();
+            var listTaskB = new List<double>() { 1.88, 2.26, 3.84, 4.55, -6.21 };
+            TaskFunction(a, b, xs, xe, dx, listTaskB, listA, listB);
```
Comment (alexthvest) [inline CourseApp/Calc.cs:19]: It's not good to run tasks in constructor

File: CourseApp/Calc.cs [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        public void CalcTaskA(double a, double b, double xs, double xe, double dx, List<double> listA)
```
Comment (alexthvest) [inline CourseApp/Calc.cs:22]: Can we move specific tasks from `Calc` class? It should not known about them

File: CourseApp/Calc.cs [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+            for (double x = xs; x <= xe; x += dx)
+            {
+                listA.Add(CalcFunc(a, b, x));
+            }
+        }
+
+        public void CalcTaskB(double a, double b, List<double> listTaskB, List<double> listB)
+        {
+            var size = listTaskB.Count;
+            for (int i = 0; i < size; i++)
```
Comment (alexthvest) [inline CourseApp/Calc.cs:39]: Use `foreach` loop

File: CourseApp/Calc.cs [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+            var listTaskB = new List<double>() { 1.88, 2.26, 3.84, 4.55, -6.21 };
+            TaskFunction(a, b, xs, xe, dx, listTaskB, listA, listB);
+        }
+
+        public void CalcTaskA(double a, double b, double xs, double xe, double dx, List<double> listA)
+        {
+            for (double x = xs; x <= xe; x += dx)
+            {
+                listA.Add(x);
+            }
```
Comment (alexthvest) [inline CourseApp/Calc.cs:27]: Do we need this? Could we add `x` in the same loop when we calculating `y`?

File: CourseApp/Calc.cs [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+        {
+            return Math.Round((Math.Pow(Math.Pow(x - a, 2), 1.0 / 3.0) + Math.Pow(Math.Abs(x + b), 1.0 / 5.0)) / Math.Pow(Math.Pow(x, 2) - Math.Pow(a - b, 2), 1.0 / 9.0), 3);
+        }
+
+        public void OutputA(List<double> listA)
+        {
+            Console.WriteLine(Colors.FgCyan("-----------Task A-----------").BgMagenta());
+            for (int i = 0; i < (listA.Count / 2); i++)
+            {
+                Console.WriteLine(Colors.FgCyan($"x = {listA[i]}  y = {listA[(listA.Count / 2) + i]}").BgMagenta()); // x - the first element, y - the first element from the middle
```
Comment (alexthvest) [inline CourseApp/Calc.cs:60]: I think tuple will be more suitable to store x and y
https://docs.microsoft.com/ru-ru/dotnet/csharp/language-reference/builtin-types/value-tuples

**Review by alexthvest — CHANGES_REQUESTED** [review]

File: CourseApp/Program.cs [hunk 11]
```diff
@@ -2,12 +2,39 @@
...
+
+            Console.WriteLine(Colors.FgCyan("-----------Task B-----------").BgMagenta());
+            OutputTask(listB);
+        }
+
+        public static void OutputTask(List<double> listTask)
+        {
+            foreach (var y in listTask)
+            {
+            Console.WriteLine(Colors.FgCyan($"y = {y}").BgMagenta());
```
Comment (alexthvest) [inline CourseApp/Program.cs:36]: Fix formatting

File: CourseApp/Calculation.cs [hunk 12]
```diff
@@ -0,0 +1,28 @@
+namespace CourseApp
+{
+    using System;
+    using System.Collections.Generic;
+
+    public class Calculation : MainFunction
```
Comment (alexthvest) [inline CourseApp/Calculation.cs:6]: Please do not use inheritance here, composition or aggregation will be more suitable

File: CourseApp/Calculation.cs [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        public List<double> CalculationTask(double a, double b, double xs, double xe, double dx, List<double> listA)
```
Comment (alexthvest) [inline CourseApp/Calculation.cs:8]: Do not mutate list, just create new in method body

**Review by alexthvest — CHANGES_REQUESTED** [review]

**Review by alexthvest — APPROVED** [review]

**Pull request merged**